### ISSUE TITLE
DownloadMgr: Better logging for proxy and hosts

### ISSUE THREAD

**Issue by HereThereBeDragons** [cvmfs/cvmfs#3616]
There seems to be intermitten availability of certain S1s and the current logs are not good enough to recognize it easily.

Curl cannot distinguish between host (s1) or proxy failure when CURLE_COULDNT_CONNECT is returned, and also it is not easy to check this from the cvmfs client

Therefore, the following logging improvements should be introduced:
- add host names (not just ips) to logging message
- during proxy failover add which host is behind it

this most likely cannot be tracked in the downloadmgr but must be tracked in each jobinfo obj

**Comment by DrDaveD** [comment]
I often find that the syslog message is too vague, saying things like "Host connection problem" but if you enable the debug log it gives a more detailed message.  Will this include the more detailed message in the syslog?

**Comment by HereThereBeDragons** [comment]
hi @DrDaveD 
the pr #3617 is already up, have a look.
but no, in particular for `host connection problem` there is no good solution. however, the pr tries to make it a bit easier to see what is going on.

the error code behind that particular error from curl is
```
CURLE_COULDNT_CONNECT (7)
Failed to connect() to host or proxy. 
```

curl itself cannot detect if the proxy or the host fails.

for this reason the pr will add the proxy name to the proxy ip and the s1 when failing over. as such it should be a bit easier to see if one of the s1 has problems - or if it is a proxy connection problem

**Comment by DrDaveD** [comment]
How about this for an example.  Set `CVMFS_HTTP_PROXY` to point to a bad host name so there's an error resolving the host name.    With client 2.11.3 I see
```
Jul  3 17:22:14 dwdosgdev cvmfs2[3713474]: (config-osg.opensciencegrid.org) (manager 'default') switching proxy from http://bogus.fnal.gov:3128 to http://131.225.188.245:3126. Reason: failed proxy
```
where if  enable the debuglog and look in there I see
```
(download) (manager 'default') failed to resolve IP addresses for bogus.fnal.gov (4 - unknown host name)    [07-03-2024 17:22:12 CDT]
```
and later there's a curl error
```
(curl) {info} Failed writing header    [07-03-2024 17:22:14 CDT]
```

Or if I connect to a host where the squid isn't running but the port is open so it gets Connection refused, I get a similar useless error in /var/log/messages but debuglog shows
```
(curl) {info} Failed to connect to 131.225.60.240 port 3128 after 0 ms: Error [07-03-2024 17:27:17 CDT]
```

People shouldn't have to enable the debuglog to find such useful information.

**Comment by HereThereBeDragons** [comment]
hi dave,
the 2nd error message you are refering to also appears in the syslog

```
    LogCvmfs(kLogDownload, kLogDebug | kLogSyslogWarn,
             "(manager '%s') failed to resolve IP addresses for %s (%d - %s)",
             name_.c_str(), host.name().c_str(), new_host.status(),
             dns::Code2Ascii(new_host.status()));
```

the path of the error message is however only called when the name resolving information is not up to date anymore. i do not know how long the TTL is for that but i guess 10-15min at least. so if you have a failure in between that time, you will not get this dedicated error message and just that the proxy failed

and the curl info is really just for debugging as this creates really way too much output. also, on the current cvmfs version it is no problem to reload the client (without remounting) and switch into debugmode.

**Comment by DrDaveD** [comment]
How about saving the last `(curl) (info)` message and including that in the error output?

**Comment by HereThereBeDragons** [comment]
i really, really think this is not what you want in production:
all those `(curl)` extra output are output given when you run curl in extra verbose debug mode.
as this is all done by callbacks you dont even know what the last messages is

and as i said before: it is easy to switch to debug mode. does not need a remount of the mountpoint.

**Comment by vvolkl** [comment]
There may be ways to get additional information to the syslog without switching to debug mode, which is not always that practical, even though it's now easier thanks to Laura. We'll try to do this in another PR though.

**Comment by HereThereBeDragons** [comment]
Closed as #3694 is the follow up issue
